internal/bgp/frr/config.go

```diff
@@ -33,20 +33,53 @@ hostname {{.Hostname}}
 ip nht resolve-via-default
 ipv6 nht resolve-via-default
 
+{{- range $community, $prefixes := .PrefixesV4ForCommunity }}
+{{- range $p := $prefixes.Elements }}
+ip prefix-list {{$community}}-v4prefixes permit {{$p}}
+{{- end}}
+{{- end}}
+{{- range $community, $prefixes := .PrefixesV6ForCommunity }}
+{{- range $p := $prefixes.Elements }}
+ip prefix-list {{$community}}-v6prefixes permit {{$p}}
+{{- end}}
+{{- end}}
+{{- range $localPref, $prefixes := .PrefixesV4ForLocalPref }}
+{{- range $p := $prefixes.Elements }}
+ip prefix-list {{$localPref}}-v4localpref-prefixes permit {{$p}}
+{{- end}}
+{{- end}}
+{{- range $localPref, $prefixes := .PrefixesV6ForLocalPref }}
+{{- range $p := $prefixes.Elements }}
+ip prefix-list {{$localPref}}-v6localpref-prefixes permit {{$p}}
+{{- end}}
```

Comment:
we are creating multiple identical prefixlist one for local and another with community why not create on prefix list and use it for both rules u can have multiple route-map rules match against the same match criteria

Reply:
That's because you may have one addresspool with a localPref and one addresspool with community (and one with both).
So the lists must be separate.

See for example https://github.com/metallb/metallb/pull/1172/files#diff-ca8f31c1976ff57d31792d0304594932a61322e8328de154fcefb10f8cde74a3R7 , where one advertisement has the community and the other one has both the community and the local pref:

```
ip prefix-list 1111:2222-v4prefixes permit 172.16.1.10/24
ip prefix-list 1111:2222-v4prefixes permit 172.16.1.11/24
ip prefix-list 2-v4localpref-prefixes permit 172.16.1.11/24
```

The .11 goes to both lists, but the 10 goes only to the community one.

Reply:
the way I see this we have prefix lst for every prefix we adv
```
ip prefix-list 172.16.1.10-v4prefixes permit 172.16.1.10/24
ip prefix-list 172.16.1.11-v4prefixes permit 172.16.1.11/24
```
then
```
route-map ..... permit 1
match 172.16.1.10-v4prefixes 
if (communit) 
  set community ...
if (locPref)
  set localPref ...  
 ```

Reply:
We have Advertisements under neighbors, so that will result in duplicating the access rules again, unless drastically changing the template.

Even so, I see both solutions are equivalent. In one case we have multiple prefixes (one for each address), in the other we have multiple rules, one for each community / local pref.

We could do an extra optimization and squash prefixes that are the same, but that would make things even more complex and not sure it's worth the effort.

Reply:
ok lets hold on any optimization for now till the 1st release out then we can refactor this further

Reply:
at some point in the future we should refrain from adding those route-map under the cover and have CR for the user to add route-map rules directly